### PR TITLE
C-4854: Support new features in Send API

### PR DESCRIPTION
**Pull request by tk26**
## Description of the change

Provide support for enhanced Send API features

Verified with a toy app talking to my dev env
Screen Shot 2022-01-31 at 01 52 24
Screen Shot 2022-01-31 at 01 52 16

Verified that the message shows up in logs as intended + that the message actually gets sent.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]()

### Development

- [ ] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v2/examples/send/send_message.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+	"github.com/trycourier/courier-go/v2"
+)
+
+func sendMessage() {
+	client := courier.CreateClient("<YOUR_AUTH_TOKEN>", nil)
+
+	requestID, err := client.SendMessage(context.Background(), courier.SendMessageRequestBody{
+		Message: map[string]interface{}{
+			"template": "my-template",
```
Comment (suhasdeshpande) [inline v2/examples/send/send_message.go:15]: is this strictly typed or similar to how things are in Javascript?

File: v2/send.go [hunk 2]
```diff
@@ -13,6 +13,12 @@ type SendBody struct {
 	Brand    string      `json:"brand,omitempty"`
 }
 
+// SendMessageRequestBody is the JSON body expected by
+// Courier's /send endpoint that accepts a message object
+type SendMessageRequestBody struct {
+	Message interface{} `json:"message"`
```
Comment (suhasdeshpande) [inline v2/send.go:19]: is this a generic placeholder? I don't read Go that well so it is a naive question

**Review by suhasdeshpande — APPROVED** [review]

File: v2/send.go (hunk 2 above)
Comment (tk26) [reply v2/send.go:19]: Go typing is shit. no way to do complex nested, unions etc. So best way to keep things going is interface{} at top level which can take in anything

Comment (tk26) [reply v2/send.go:19]: so to ans your ques yep a generic placeholder

File: v2/examples/send/send_message.go (hunk 1 above)
Comment (tk26) [reply v2/examples/send/send_message.go:15]: top level Message would be strict. but that bag would have anything.

**Pull request merged**